src/_guides/language/language-tour_migrated.md

```diff
@@ -363,6 +363,51 @@ assert(lineCount == null);
   _condition_ is false. For details, see [Assert](#assert).
 {{site.alert.end}}
 
+
+### Late variables
+
+Dart 2.12 added the `late` modifier, which has two use cases:
+
+* To tell Dart that a non-nullable variable is set before it's used.
+* To lazily initialize a variable.
+
+Often Dart's control flow analysis can detect when a non-nullable variable
+is set to a non-null value before it's used,
+but sometimes analysis fails.
```

Comment:
This is generally true. But there's a fairly big split between how local variables are analyzed and how all other variables are. Even with locals there are edge cases where the flow analysis fails. But for fields and top-level variables, there is no flow analysis at all. If you don't initialize it at the declaration (or initializer list for instance fields), it's an error.